packages/pg-query-stream/test/async-iterator.ts

```diff
@@ -122,9 +122,11 @@ if (!process.version.startsWith('v8')) {
       const pool = new pg.Pool({ max: 1 })
       const client = await pool.connect()
 
+      /* eslint-disable @typescript-eslint/no-unused-vars */
       for await (const _ of client.query(new QueryStream('select TRUE', [], { highWaterMark: 1 }))) break
```

Comment:
I'm not sure how this line is different to:

```
await client.query(new QueryStream('select TRUE', [], { highWaterMark: 1 }))
```

Reply:
The difference is that it actually consumes the async iterable. `client.query` returns the `QueryStream` when it’s passed one, so plain `await` would do nothing.

Reply:
Thanks!

> it actually consumes the async iterable

I think "consumes" is a strong word - it looks like `break` means it just takes the first item from the query stream.  It looks analogous to this:

```js
let calls=0;
async function* fakeQueryStream() {
  yield ++calls;
  yield* fakeQueryStream();
}
for await(const _ of fakeQueryStream()) break;
console.log('calls:', calls);
```

which will print:

```
calls: 1
```

-----

This suggests that the current code

```js
for await (const _ of client.query(new QueryStream('select TRUE', [], { highWaterMark: 1 }))) break
```

is the equivalent of:

```js
await client.query(new QueryStream('select TRUE', [], { highWaterMark: 1 }))[Symbol.asyncIterator]().next()
```

-----

To consume the stream, I think `break` should be replaced with `{}` or `continue`.

Reply:
There’s only one row, but either asserting that or using an empty loop body seems good, yes.

Reply:
Hmm the test title is "supports breaking with low watermark", so maybe it's explicitly testing something to do with `break`...??  Anyway, not directly relevant to this PR.